csharp/src/Ice/ConnectionI.cs

```diff
@@ -1447,24 +1450,57 @@ internal void sendHeartbeat()
         {
             if (isActiveOrHolding())
```

Comment:
I find it would be simpler and clearer to just remove `isActiveOrHolding` (which itself unnecessarily locks the connection). The first if block checks the state, the second could also check the state. The `idleCheck()`  implementation could do the same.

Reply:
Ok, will remove in a follow-up PR.